`dis.dis` has incorrect annotation

### Env
- Python 3.7.2
- MyPy 0.700

### Problem Details

Following code

```Python
import dis


def f(a: int) -> int:
    return a + 2


dis.dis(f)
```

MyPy will report 

```
x.py:8: error: Argument 1 to "dis" has incompatible type "Callable[[int], int]"; expected "Union[MethodType, FunctionType, CodeType, type, str, bytes]"
```

It seems MyPy does not parse it as `types.FunctionType`, I'm not sure it's a bug of MyPy or typeshed. I check the code of `dis.dis` (https://github.com/python/cpython/blob/9a3ffc0492d1310ead9ce8f5ee678c26b20a338d/Lib/dis.py#L34-L77), I think implementing a `Protocol` is a workaround.

--- a/stdlib/2and3/dis.pyi
+++ b/stdlib/2and3/dis.pyi
@@ -14,7 +14,9 @@ if sys.version_info >= (3, 4):
 if sys.version_info >= (3, 6):
     from opcode import hasnargs as hasnargs
 
-_have_code = Union[types.MethodType, types.FunctionType, types.CodeType, type]
+# Strictly this should not have to include Callable, but mypy doesn't use FunctionType
+# for functions (python/mypy#3171)
+_have_code = Union[types.MethodType, types.FunctionType, types.CodeType, type, Callable[..., Any]]
 _have_code_or_string = Union[_have_code, str, bytes]
 
 

--- a/stdlib/2and3/dis.pyi
+++ b/stdlib/2and3/dis.pyi
@@ -1,4 +1,4 @@
-from typing import List, Union, Iterator, Tuple, Optional, Any, IO, NamedTuple, Dict
+from typing import Callable, List, Union, Iterator, Tuple, Optional, Any, IO, NamedTuple, Dict
 
 import sys
 import types

